### PR TITLE
Add low priority XPU Stream

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #143849
* #143799
* #141123
* __->__ #141119
* #142347

# Motivation
Due to the potential for the external SYCL queue to have a low priority, we need to support the low-priority SYCL queue for native XPU Streams to maintain consistency.

cc @gujinghui @EikanWang @fengyuan14

### CODE REVIEW

**Review by gujinghui — APPROVED** [review]

File: c10/xpu/XPUStream.cpp [hunk 1]
```diff
@@ -232,14 +261,11 @@ XPUStream getStreamFromPool(const int priority, DeviceIndex device) {
     device = c10::xpu::current_device();
   }
   check_device_index(device);
-  TORCH_CHECK(
```
Comment (albanD) [inline c10/xpu/XPUStream.cpp:235]: Should we remove this check from all implementations (in particular the CUDA one)?
That way the python `torch.Stream(priority=1)`  would mean the lowest priority for both? And in particular using -10/10 like you do in your test would be a good cross-device strategy to get the highest/lowest priority?

**Review by albanD — APPROVED** [review]
Change sounds ok, question about making the API easier to share.

File: c10/xpu/XPUStream.cpp (hunk 1 above)
Comment (guangyey) [reply c10/xpu/XPUStream.cpp:235]: OK, I will remove CUDA's check in another PR.

File: torch/xpu/streams.py [hunk 2]
```diff
@@ -21,9 +21,10 @@ class Stream(torch._C._XpuStreamBase):
         device(torch.device or int, optional): a device on which to allocate
             the stream. If :attr:`device` is ``None`` (default) or a negative
             integer, this will use the current device.
-        priority(int, optional): priority of the stream, should be 0 or
-            negative, where negative numbers indicate higher priority. By default,
-            streams have priority 0.
+        priority(int, optional): priority of the stream, which can be positive, 0, or negative.
```
Comment (guangyey) [inline torch/xpu/streams.py:24]: cc @albanD , I updated this description.

Comment (guangyey) [reply torch/xpu/streams.py:24]: > cc @albanD , I updated this description. Please help review again.

File: torch/xpu/streams.py [hunk 3]
```diff
@@ -21,9 +21,10 @@ class Stream(torch._C._XpuStreamBase):
...
+            A lower number indicates a higher priority. By default, the priority is set to 0.
+            If the value falls outside of the allowed priority range, it will automatically be
+            mapped to the nearest valid priority(either lowest or highest).
```
Comment (albanD) [inline torch/xpu/streams.py:27]:
```suggestion
            mapped to the nearest valid priority (lowest for large positive numbers or highest for large negative numbers).
```

**Review by albanD — APPROVED** [review]
Sounds good

File: torch/xpu/streams.py (hunk 3 above)
Comment (guangyey) [reply torch/xpu/streams.py]: Thanks, updated.

**Pull request closed without merging**